debugging-contd/problem-set-debugging-contd.md

```diff
@@ -160,63 +121,63 @@ The for loop is now iterating over each horizontal position in each inner list.
 * title: Debugging Continued
-<!--Other optional fields (checkpoints only) -->
-<!--`points: 1`: the number of points for scoring as a checkpoint-->
-<!--`topics: python, pandas`: the topics for analyzing points-->
-
 ##### !question
 
-Moira's code produces this output:
-
-![Moira's code output.  The first generation has three vertical alive cells.  The second generation has no alive cells.](../assets/debugging-contd_ps-output-bug.png)  
-_Fig. Moira's code output._
+Dahlia's code produces this output when ran in test mode:
+
+```py
+Please enter r to run or t to test => t
+Traceback (most recent call last):
+  File "/home/runner/EncodeAndDecodeGeneticSequencesStarter/main.py", line 17, in <module>
+    tests()
+  File "/home/runner/EncodeAndDecodeGeneticSequencesStarter/sequence_tests.py", line 16, in tests
+    part_one_tests()
+  File "/home/runner/EncodeAndDecodeGeneticSequencesStarter/sequence_tests.py", line 21, in part_one_tests
+    dna_tests()
+  File "/home/runner/EncodeAndDecodeGeneticSequencesStarter/sequence_tests.py", line 27, in dna_tests
+    assert result == DNA_ENCODED
+AssertionError
+```
 
-What is the bug that Moira sees now?  What is the expected output?  What is the code doing?
+Answer the following question:
+1. What is the bug that Dahlia sees now?  
+2. What is the expected output?  
+3. What is the code doing?
 
 ##### !end-question
-
-<!--optional-->
 ##### !explanation
 
-The first generation has three living vertical cells.  The second generation should have three living horizontal cells.  Moira's code produces a second generation where all of the cells are dead.
+1. Dahlia is seeing a mismatch between the string that `encode_strand` is returning and the expected string for the test `DNA_ENCODED`.
 
-##### !end-explanation
+2. The expected output should include all bases listed in the genetic sequence we're compressing.
 
-### !end-challenge
+3. If we use the debugger or print statements to look at the values of `result` and `dna_encoded1`, we'll see that the last base in a sequence and its count is not being added to the compressed sequence stored in `result`. 
 
+##### !end-explanation
+### !end-challenge
 <!-- prettier-ignore-end -->
-<!--END CHALLENGE-->
 
-<!--BEGIN CHALLENGE-->
 <!-- prettier-ignore-start -->
-
 ### !challenge
-
 * type: paragraph
 * id: f9f85ba0-3390-46c0-9244-38b1bffa6821
 * title: Debugging Continued
-
 ##### !question
 
-Moira wants to get her code working, but before she can do that she needs to understand why her code is doing what it's doing. 
+Dahlia wants to get her code working, but before she can do that she needs to understand why her function `encode_strand` is behaving unexpectedly. 
 
 Take a few minutes and use some debugging techniques to find out why her code is behaving the way it is.
 
 List a few of the things you tried and what you discovered.
 
 ##### !end-question
-
-<!--optional-->
 ##### !explanation
 
 Some examples of things to try:
```

Comment:
Think we need to update the explanations for this question (lines 185-187).

- Add print statements to `encode_strand` inside the for-loop to see what is getting appended to the list called `encoding` with each iteration.